bot/utils/pagination.py

```diff
@@ -0,0 +1,389 @@
+
+            ii) remaining_words: the words in `line` which exceed `max_chars`.
+        """
+        reduced_words = line[:max_chars].rsplit(" ", 1)[0]
+        remaining_words = line.replace(reduced_words, "")
+
+        return (
+            reduced_words + "..." if remaining_words else "",
+            CONTINUATION_HEADER + remaining_words if remaining_words else None,
+        )
+
+    @classmethod
+    async def paginate(
+        cls,
+        lines: t.List[str],
+        ctx: Context,
+        embed: discord.Embed,
+        prefix: str = "",
+        suffix: str = "",
+        max_lines: t.Optional[int] = None,
+        max_size: int = 500,
+        scale_to_size: int = 2000,
+        empty: bool = False,
+        restrict_to_user: User = None,
+        timeout: int = 300,
+        footer_text: str = None,
+        url: str = None,
+        allow_empty_lines: bool = False,
+    ) -> t.Optional[discord.Message]:
+        """
+        Use a paginator and set of reactions to provide pagination over a set of lines.
+
+        When used, this will send a message using `ctx.send()` and apply the pagination reactions,
+        to control the embed.
+
+        Pagination will also be removed automatically if no reaction is added for `timeout` seconds.
+
+        The interaction will be limited to `restrict_to_user` (ctx.author by default) or
+        to any user with a moderation role.
+
+        Example:
+        >>> people = ["Guido van Rossum", "Linus Torvalds", "Gurkbot", "Bjarne Stroustrup"]
+        >>> e = discord.Embed()
+        >>> e.set_author(name="Ain't these people just awesome?")
+        >>> await LinePaginator.paginate(people, ctx, e)
+        """
+
+        def event_check(reaction_: discord.Reaction, user_: discord.Member) -> bool:
+            """Make sure that this reaction is what we want to operate on."""
+            return (
+                # Conditions for a successful pagination:
+                all(
+                    (
+                        # Reaction is on this message
+                        reaction_.message.id == message.id,
+                        # Reaction is one of the pagination emotes
+                        str(reaction_.emoji) in PAGINATION_EMOJI,
+                        # Reaction was not made by the Bot
+                        user_.id != ctx.bot.user.id,
+                        # The reaction was by a whitelisted user
+                        user_.id == restrict_to_user.id,
+                    )
+                )
+            )
+
+        paginator = cls(
+            prefix=prefix,
+            suffix=suffix,
+            max_size=max_size,
+            max_lines=max_lines,
+            scale_to_size=scale_to_size,
+        )
+        current_page = 0
+
+        # If the `restrict_to_user` is empty then set it to the original message author.
+        restrict_to_user = restrict_to_user or ctx.author
+
+        if not lines:
+            if allow_empty_lines:
```

Comment:
error out if `allow_empty_lines` is `False`.
```suggestion
            if not allow_empty_lines:
```